File: packages/javascript/lib/word/randEmotion.ts
import type { RandThemedWordOptions, WordDetail } from '../_types/global.js';
import { themedWord } from './wordGenerator.js';

/**
 * What someone feels, from joy to a quiet regret.
 *
 * `randWord({ theme: 'emotion' })` with the theme already chosen; every other
 * option is the same.
 *
 * @example
 * randEmotion({ language: 'ko', count: 3 }); // ['그리움', '설렘', '안도']
 * randEmotion({ language: 'en', count: 3 }); // ['Longing', 'Delight', 'Relief']
 */
export function randEmotion(options?: RandThemedWordOptions & { output?: 'value' }): string[];
/**
 * Emotion words with the language and theme each one came from.
 *
 * @example
 * randEmotion({ language: 'ko', output: 'detail' });
 * // [{ word: '그리움', language: 'ko', theme: 'emotion' }]
 */
export function randEmotion(options: RandThemedWordOptions & { output: 'detail' }): WordDetail[];
export function randEmotion(options: RandThemedWordOptions = {}): string[] | WordDetail[] {
	return themedWord('emotion', options);
}

File: packages/dart/lib/src/word/rand_emotion.dart
import 'dart:math';

import 'package:randino/src/types.dart';
import 'package:randino/src/word/rand_word.dart';

/// What someone feels, from joy to a quiet regret.
///
/// [randWord] with the theme already chosen; every other parameter is the same.
/// For the detail form, pass [WordTheme.emotion] to `randWordDetails` — Dart has
/// no overloads, and a second function per theme would be a second wall of them.
///
/// ```dart
/// randEmotion(language: WordLanguage.ko, count: 3); // [그리움, 설렘, 안도]
/// randEmotion(language: WordLanguage.en, count: 3); // [Longing, Delight, Relief]
/// ```
List<String> randEmotion({
  WordLanguage? language,
  int count = 1,
  RandRealism realism = RandRealism.real,
  RandVocabulary vocabulary = RandVocabulary.full,
  int? minLength,
  int? maxLength,
  String? startsWith,
  bool unique = false,

  /// Where the randomness comes from: `Random.secure()` for a value nobody may
  /// predict, `Random(42)` for one that has to come out the same every run.
  Random? random,
}) => randWord(
  language: language,
  theme: WordTheme.emotion,
  count: count,
  realism: realism,
  vocabulary: vocabulary,
  minLength: minLength,
  maxLength: maxLength,
  startsWith: startsWith,
  unique: unique,
  random: random,
);
